City-of-Helsinki/linkedcomponents-ui
# LINK-183, LINK-184, LINK-185, LINK-186, LINK-187, LINK-188, LINK-189 | Event list view

## Description
Add footer to the page layout
Pagination component
Event card component to be used on the 
Table view to display list of the own events of signed user
Event card view to display list of the own events of signed user

## Closes
[LINK-183](https://helsinkisolutionoffice.atlassian.net/browse/LINK-183)
[LINK-184](https://helsinkisolutionoffice.atlassian.net/browse/LINK-184)
[LINK-185](https://helsinkisolutionoffice.atlassian.net/browse/LINK-185)
[LINK-186](https://helsinkisolutionoffice.atlassian.net/browse/LINK-186)
[LINK-187](https://helsinkisolutionoffice.atlassian.net/browse/LINK-187)
[LINK-188](https://helsinkisolutionoffice.atlassian.net/browse/LINK-188)
[LINK-189](https://helsinkisolutionoffice.atlassian.net/browse/LINK-189)

## Screenshot
<img width="1434" alt="Screenshot 2021-01-12 at 12 26 08" src="https://user-images.githubusercontent.com/24706814/104302898-0a716100-54d2-11eb-901a-6ef9dfdc4711.png">

<img width="1440" alt="Screenshot 2021-01-12 at 12 27 01" src="https://user-images.githubusercontent.com/24706814/104302952-1e1cc780-54d2-11eb-908c-e137cd4f51fd.png">

## Testing
1. Remove comment from line 116 `(apikey: '50381be7-fef2-4783-b181-3181f6492f3f',)` on domain/app/apollo/apolloClient.ts file to use apikey for authentication 
2. Make sure that right url is set for LE by setting `REACT_APP_LINKED_EVENTS_URL=http://localhost:8000/v1` in .env.development.local
2. Setup local LE backend and set apikey for some data source
- Clone LE backend repository from https://github.com/City-of-Helsinki/linkedevents
- Add a local_settings.py file and add following lines to it to support apikey in request header
```
DEBUG = True
CORS_ALLOW_HEADERS = [
    'apikey',
    'authorization',
    'content-type',
    'dnt',
    'origin',
    'user-agent',
    'x-csrftoken',
    'x-requested-with',
]
```
- Start docker container: `docker-compose up`. Make sure ports 5432 or 8000 are not already in use
- In other terminal tab open django bash to run importers: `docker-compose exec django bash`
- At django bask run importer scripts:
```
python manage.py event_import yso --all
python manage.py add_helsinki_audience
python manage.py add_helsinki_topics
python manage.py event_import tprek --places
python manage.py import_organizations https://api.hel.fi/paatos/v1/organization/ -s helsinki:ahjo
```
- Add apikey `50381be7-fef2-4783-b181-3181f6492f3f` to one of the datasource .e.g http://localhost:8000/admin/events/datasource/helsinki/change/
<img width="1438" alt="Screenshot 2021-01-12 at 12 45 54" src="https://user-images.githubusercontent.com/24706814/104304524-2544d500-54d4-11eb-89cc-d9b78daa7d6c.png">
